woocommerce/woocommerce
# Fix / PHP Fatal Error Triggered by Invalid Offset Type in wc-cart-functions.php #41546

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





This PR addresses a fatal error triggered by an illegal offset type in isset  within the wc-cart-functions.php.
Includes some refactoring to the method.

Closes #41546



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

Dev-facing testing instructions

1. Set the chosen method to an array `WC()->session->set( 'chosen_shipping_methods', array( $key => array('oops')));` just before `$chosen_methods = WC()->session->get( 'chosen_shipping_methods' );`
2. Verify no fatal error is produced



### Changelog entry




<details>

#### Significance




#### Type




#### Message 
Added validation to prevent fatal error in wc-cart-functions.php.

#### Comment 

</details>
